window.c

```diff
@@ -486,17 +511,44 @@ void win_draw_rect(win_t *win, int x, int y, int w, int h, bool fill, int lw,
 		XDrawRectangle(win->env.dpy, win->buf.pm, gc, x, y, w, h);
 }
 
-void win_set_title(win_t *win, const char *title)
+void win_set_title(win_t *win, const char *path)
 {
-	XStoreName(win->env.dpy, win->xwin, title);
-	XSetIconName(win->env.dpy, win->xwin, title);
+	char *title, *suffix="";
+
+	/* Return if window is not ready yet, otherwise we get an X fault. */
+	if (win->xwin == None)
+		return;
+
+	/* Get title suffix type from X-resources. Default: BASE_CDIR. */
+	suffix = estrdup(path);
+	switch (win->title_suffixmode) {
+		case CFILE:
+			win->title_suffix = suffix;
+			break;
+		case BASE_CFILE:
+			win->title_suffix = basename(suffix);
+			break;
+		case SUFFIXMODE_COUNT: // Never happens
+		case EMPTY:
+			win->title_suffix = "";
+			break;
+	}
 
+	/* Some ancient WM's that don't comply to EMWH (e.g. mwm) only use WM_NAME for
+	 * the window title, which is set by XStoreName below. */
+	size_t prefixLen = strlen(win->title_prefix);
+	size_t suffixLen = strlen(win->title_suffix);
+	const char *separator = (suffixLen == 0 || prefixLen == 0) ? "" : " - ";
```

Comment:
This should be removed. If the user wants a seperator they can just append it to the end of prefix, i.e `"sxiv -"`.